crates/sui-core/src/scoring_decision.rs

```diff
@@ -0,0 +1,322 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::authority::AuthorityMetrics;
+use crate::math::median;
+use arc_swap::ArcSwap;
+use narwhal_types::ReputationScores;
+use std::collections::HashMap;
+use std::sync::Arc;
+use sui_types::base_types::AuthorityName;
+use sui_types::committee::Committee;
+use tracing::{debug, info, warn};
+
+const MAD_DIVISOR: f64 = 0.7;
```

Comment:
These parameters, as well as the number of rounds for recalculating the NW scores should probably be part of config. Could you add a TODO and migrate these to a config in a followup PR?